### PR TITLE
Add HL FOFB IOC and devices

### PR DESCRIPTION
**Pull request by anacso17**
This PR:
- adds new FOFB related devices: BPMLogicalTrigger, FOFBCtrlDCC, BPMDCC, FOFBCtrlRef, FamFOFBControllers,     FamFastCorrs and HLFOFB;
- updates PwrSupplyFC device according to new fast corrector IOC version;
- adds first working version of high level FOFB IOC;
- updates cycle procedure and PSDiag IOC according to new fast corrector IOC version;
- adds si_fastorbcorr_respm configuration type;
- updates si_orbit configuration template according to new fast corrector IOC version;
- updates version.

Related PRs: 
- https://github.com/lnls-sirius/machine-applications/pull/257 
- https://github.com/lnls-sirius/docker-machine-applications/pull/37)
- https://github.com/lnls-sirius/hla/pull/562

### CODE REVIEW

File: siriuspy/siriuspy/devices/fofb.py [hunk 1]
```diff
@@ -0,0 +1,883 @@
...
+        """Init."""
+        # check if device exists
+        if devname not in self.DEVICES:
+            raise NotImplementedError(devname)
+
+        # call base class constructor
+        super().__init__(devname, properties=FOFBCtrlRef._properties)
+
+    @property
+    def ref(self):
```
Comment (xresende) [inline siriuspy/siriuspy/devices/fofb.py:47]: @anacso17 , `reforbx`, `reforby`, `reforb` look more intuitive for me. are we using `ref`in other devices?

File: siriuspy/siriuspy/devices/fofb.py [hunk 2]
```diff
@@ -0,0 +1,883 @@
...
+        # BPM DCCs and triggers
+        bpmnames = _BPMSearch.get_names({'sec': 'SI', 'dev': 'BPM'})
+        bpmids = _np.roll(
+            _np.array([i-1 if i % 2 == 1 else i for i in range(160)]), -1)
+        self._bpm_dccs, self._bpm_trgs, self._bpm_ids = dict(), dict(), dict()
+        for idx, bpm in enumerate(bpmnames):
+            self._bpm_ids[bpm] = bpmids[idx]
+            self._bpm_dccs[bpm] = BPMDCC(bpm)
+            for trig in self.BPM_TRIGS_IDS:
+                trigname = bpm+':TRIGGER' + str(trig)
```
Comment (xresende) [inline siriuspy/siriuspy/devices/fofb.py:214]: missing spaces arounf `+` operator.

File: siriuspy/siriuspy/fofb/csdev.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+    GlobIndiv = _csdev.Const.register('GlobIndiv', _et.GLOB_INDIV)
+    UseRF = _csdev.Const.register('UseRF', _et.DSBL_ENBL)
+    MeasRespMatCmd = _csdev.Const.register('MeasRespMatCmd', _et.MEAS_RMAT_CMD)
+    MeasRespMatMon = _csdev.Const.register('MeasRespMatMon', _et.MEAS_RMAT_MON)
+
+    def __init__(self):
+        """Class constructor."""
+
+        # device names and nicknames
+        self.bpm_names = _BPMSearch.get_names({'sec': 'SI', 'dev': 'BPM'})
```
Comment (xresende) [inline siriuspy/siriuspy/fofb/csdev.py:55]: I think we should not use public attributes like `self.bpm_names` and others. we should use private attributes such as `self._bpm_names`and access to tho them with property and setter decoratored functions. This way all class interface is explicit and one does not to look at the class method implementation (like `__init__`) so find out all class interface.

File: siriuspy/siriuspy/fofb/csdev.py [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+        self.reforb_fname = _os.path.join(ioc_fol, 'reforbit.orb')
+        self.respmat_fname = _os.path.join(ioc_fol, 'respmat.respmat')
+
+        # reforb and matrix parameters
+        self.reforb_size = self.nr_bpms
+        self.matrix_size = self.nr_corrs * (2 * self.nr_bpms)
+        self.nr_svals = min(self.nr_corrs, 2 * self.nr_bpms)
+        self.circum = 518.396  # in meter
+        self.harm_number = 864
+        self.rev_per = self.circum / 299792458  # in seconds
```
Comment (xresende) [inline siriuspy/siriuspy/fofb/csdev.py:89]: I think we sould use `mathphys` for this constants

File: siriuspy/siriuspy/fofb/csdev.py [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+        ioc_fol = _os.path.join(
+            '/home', 'sirius', 'iocs-log', 'si-ap-fofb', 'data')
+        self.reforb_fname = _os.path.join(ioc_fol, 'reforbit.orb')
+        self.respmat_fname = _os.path.join(ioc_fol, 'respmat.respmat')
+
+        # reforb and matrix parameters
+        self.reforb_size = self.nr_bpms
+        self.matrix_size = self.nr_corrs * (2 * self.nr_bpms)
+        self.nr_svals = min(self.nr_corrs, 2 * self.nr_bpms)
+        self.circum = 518.396  # in meter
```
Comment (xresende) [inline siriuspy/siriuspy/fofb/csdev.py:87]: can we not get these lattice parameters from somewhere else? the main definition for these global parameters should not be within a specific class.

**Review by xresende — APPROVED** [review]

File: siriuspy/siriuspy/devices/fofb.py (hunk 1 above)
Comment (anacso17) [reply siriuspy/siriuspy/devices/fofb.py:47]: We use `refx` and `refy`in SOFB device

File: siriuspy/siriuspy/fofb/csdev.py (hunk 4 above)
Comment (anacso17) [reply siriuspy/siriuspy/fofb/csdev.py:89]: We are not actually using these constants in this IOC, I will remove them

File: siriuspy/siriuspy/fofb/csdev.py (hunk 5 above)
Comment (anacso17) [reply siriuspy/siriuspy/fofb/csdev.py:87]: These constants are defined in a few places in siriuspy and siriusla. I think these constants are defined in packages like pymodels, and the idea of defining them here maybe is not to create dependencies in siriuspy

File: siriuspy/siriuspy/devices/fofb.py (hunk 2 above)
Comment (anacso17) [reply siriuspy/siriuspy/devices/fofb.py]: done!

File: siriuspy/siriuspy/fofb/csdev.py (hunk 4 above)
Comment (anacso17) [reply siriuspy/siriuspy/fofb/csdev.py]: done!

**Review by murilobalves — APPROVED** [review]

**Pull request merged**